### ISSUE TITLE
Pick Worksheet Id

### ISSUE THREAD

**Issue by smalos** [jfmonteil/Pentaho-Google-Sheet-Plugin#2]
The plugin just works fine for me. Thanks a lot!

A small issue that already existed in previous versions. If you pick a Worksheet Id and cancel the dialog (without picking a value), an error is raised:

> Argument cannot be null

You could try this fix: Add esd.open() !=null to check if the dialog was cancelled.

https://github.com/jfmonteil/Pentaho-Google-Sheet-Plugin/blob/0bf2ea63abf4dbf3b226b229d9ebec865e818764/src/main/java/org/pentaho/di/ui/trans/steps/pentahogooglesheets/PentahoGoogleSheetsPluginInputDialog.java#L515

```
if (esd.open() !=null) {
	if (esd.getSelectionIndeces().length > 0) {
		selectedSpreadsheet = esd.getSelectionIndeces()[0];
		File spreadsheet = spreadsheets.get(selectedSpreadsheet);
		spreadsheetKey.setText(spreadsheet.getId());
	} else {
		spreadsheetKey.setText("");
	}
}
```

The same goes for the Spreadsheet Key dialog.

**Comment by jfmonteil** [comment]
Thank you for your feedback, I have tryed applying the fix without success. Let me try again ;) Any more ideas welcome !

**Comment by smalos** [comment]
Just a shot in the dark: Check if selectedSpreadsheet !=null)

```
if (esd.open() !=null) {
	if (esd.getSelectionIndeces().length > 0) {
		selectedSpreadsheet = esd.getSelectionIndeces()[0];
                if (selectedSpreadsheet !=null){
		    File spreadsheet = spreadsheets.get(selectedSpreadsheet);
		   spreadsheetKey.setText(spreadsheet.getId());
                }
	} else {
		spreadsheetKey.setText("");
	}
}
```

**Comment by jfmonteil** [comment]
Fix delivered as 3.1, removing all depreciated classes seems to have had an effect

**Comment by jfmonteil** [comment]
3.1 available in 1 minute :)

**Comment by smalos** [comment]
Thanks, the error is gone with the latest update.
